plugins/modules/keycloak_realm_localization.py

```diff
@@ -0,0 +1,391 @@
+    auth_client_id: admin-cli
+    auth_keycloak_url: https://auth.example.com/auth
+    token: TOKEN
+    parent_id: my-realm
+    locale: fi
+    state: present
+    overrides:
+      - key: app.title
+        value: "Sovellukseni"
+  delegate_to: localhost
+
+- name: Remove all overrides for locale "de"
+  community.general.keycloak_realm_localization:
+    auth_client_id: admin-cli
+    auth_keycloak_url: https://auth.example.com/auth
+    auth_realm: master
+    auth_username: USERNAME
+    auth_password: PASSWORD
+    parent_id: my-realm
+    locale: de
+    state: absent
+  delegate_to: localhost
+
+- name: Dry run - see what would change for locale "en"
+  community.general.keycloak_realm_localization:
+    auth_client_id: admin-cli
+    auth_keycloak_url: https://auth.example.com/auth
+    auth_realm: master
+    auth_username: USERNAME
+    auth_password: PASSWORD
+    parent_id: my-realm
+    locale: en
+    state: present
+    overrides:
+      - key: greeting
+        value: "Hello again"
+  check_mode: true
+  delegate_to: localhost
+"""
+
+RETURN = r"""
+msg:
+  description: Human-readable message about what action was taken.
+  returned: always
+  type: str
+end_state:
+  description:
+    - Final state of localization overrides for the locale after module execution.
+    - Contains the C(locale) and the list of C(overrides) as key/value items.
+  returned: on success
+  type: dict
+  contains:
+    locale:
+      description: The locale code affected.
+      type: str
+      sample: en
+    overrides:
+      description: The list of overrides that exist after execution.
+      type: list
+      elements: dict
+      sample:
+        - key: greeting
+          value: Hello
+        - key: farewell
+          value: Bye
+diff:
+  description:
+    - When run with C(--diff), shows the before and after structures
+      for the locale and its overrides.
+  returned: when supported and requested
+  type: dict
+  contains:
+    before:
+      description: State of localization overrides before execution
+      type: dict
+      sample:
+        - key: greeting
+          value: Hello
+        - key: farewell
+          value: Bye
+    after:
+      description: State of localization overrides after execution
+      type: dict
+      sample:
+        - key: greeting
+          value: Hello
+        - key: farewell
+          value: Bye
+"""
+
+from ansible_collections.community.general.plugins.module_utils.identity.keycloak.keycloak import KeycloakAPI, \
+    keycloak_argument_spec, get_token, KeycloakError
+from ansible.module_utils.basic import AnsibleModule
+from copy import deepcopy
+
+
+def _normalize_overrides_from_api(current):
+    """
+    Accept either:
+      - dict: {'k1': 'v1', ...}
+      - list of dicts: [{'key': 'k1', 'value': 'v1'}, ...]
+    Return a sorted list of {'key', 'value'}.
+
+    This helper provides a consistent shape for downstream comparison/diff logic.
+    """
```

Comment:
from the rest of the code, this function will only receive the output of `kc.get_localization_values()`, which is also part of this PR and always returns a `dict`. So why is this `list` case here to begin with?

#aifeelings

Reply:
Ah, true. Actually not so much as AI as oversight after code refactoring. There had been a revision in which this was useful. AI was used only at the end to point out possible bugs and to generate documentation (thanks for pointing out those links above, missed them during my search) and tests. But the code is written by me through modification of another module. Only issue is this is my first ansible module ever :).  Same with the value check below.